### DISCUSSION TITLE
Is a flow deck sufficient for implementing autonomous behavior?

### DISCUSSION THREAD

**Discussion by noahra** [bitcraze/discussions#616] [Q&A] [answered]
I watched a work shop on youtube where autonomous behavior was implemented on one of your drones. The dude there used both a flow deck and a multiranger deck. Will I need both of those if I want to implement autonomous behavior?

**Comment by hmllr** [comment, marked as answer]
Hi :)
Autonomous behavior can describe a very wide range of applications. 
If you define autonomous just as acting without human control, you could theoretically  achieve something like this with only the flow deck - it would be very very limited though, you could for example follow a height change on the floor. As you cannot sense anything in your surroundings but the distance to the closest object below you and the x and y movement the environment would have to be very very specific for it to work.
In robotics, you split define autonomy in three parts: (a) perception, you have to sense the surroundings (b) decision-making, you have to use the perception to decide what to do and (c) actuation, you need to be capable to execute this decision.
Of course you could fit some example with only the flow deck into this definition as well, as described earlier - however, adding a multi-ranger gives you perception in front/back/left/right/up direction as well, and with this opens many more possibilities for different autonomous applications as wall following (see our demo example: https://github.com/bitcraze/crazyflie-firmware/tree/master/examples/demos/app_wall_following_demo it is not documented, but it is basically just following the first wall it finds) or of course many other tasks you can implement yourself :)
Hope this clarifies it,
Hanna

**Reply by CalvinKoung** [reply]
Hi, If i want to create autonomous behavious in my drone. Can I use Z-ranger deck v2 and loco position system together to achieve this goal ?  And I saw the Z-ranger deck v2 and flow deck this two deck function describe is similiar. Can I know specific different in two deck ? thanks.

**Reply by ataffanel** [reply]
If you have the loco positioning system it is enough to fly autonomously. The Z-ranger will help with Z position estimation. The flow deck will help with X-Y stability and Z estimation: the flow deck has an optical flow sensor that can estimate the X-Y velocity. Both the Flow deck and Z ranger deck have a time-of-flight optical ranging sensor to estimate the Crazyflie's height over the floor.

**Reply by CalvinKoung** [reply]
OK, thanks your reply.

**Reply by knmcguire** [reply]
Just a note for next time @CalvinKoung, if you have any further questions, please start a completely new discussion please. This will be easier to handle for my colleagues and the community, since this thread was already considered answered

**Reply by CalvinKoung** [reply]
Sorry, I though this question similar with this topic so I asked here. Next time, I will start a new discussion.

**Reply by knmcguire** [reply]
ah yeah no worries! we will refer you to the right threads in case it is too similar, but your question had some differences though:) Just good to know for next time!

**Reply by CalvinKoung** [reply]
Ok, I got it. thanks your help
